### ISSUE TITLE
Cdktf creating a Zombie process named "[node] <defunct>" on each cdktf cli execution.

### ISSUE THREAD

**Issue by surajgawas110** [hashicorp/terraform-cdk#2534]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**cdktf & Language Versions**
cdktf version = 0.15.1
Terraform v1.3.7 on linux_amd64
Language : Java  - openjdk-11

**Issue :** Whenever we try to execute Cdktf code to deploy or destroy resources, a zombie process is being created for each run, no matter what the execution status of terraform pass or fail.
We are executing this cdktf code within a Node docker container with version 19.4.0

**Impact :** Lot of zombie processes are being created for each single run of cdktf commands. Thereby impacting application performance within container as well as host.

**Before cdktf execution we had 1 zombie process from earlier builds as we can see in below screenshot.**
![image](https://user-images.githubusercontent.com/48012652/214515625-d3051e2b-16fa-4650-a341-16eca3bc83fe.png)

**After cdktf execution you can see new highlighted zombie process is created.**
![image](https://user-images.githubusercontent.com/48012652/214515775-d91354b9-e5b6-4efe-bd03-266a28da4d8a.png)

**Attaching terraform code in java as well as destroy command for the same.**
[Main.java.txt](https://github.com/hashicorp/terraform-cdk/files/10497836/Main.java.txt)

**Command to execute**
cdktf destroy AwsStack-000 --auto-approve --app "mvn --settings ./settings.xml -e -q compile exec:java  -Daws_access_key_id=ACCESS_KEY -Daws_secret_access_key=SECRET -DrunId=000 -DavailablePort=55006 -Dplatform=ECS -Dregions=ap-southeast-2,sa-east-1"

Attaching dockerfile for container debugging purpose
[Dockerfile.txt](https://github.com/hashicorp/terraform-cdk/files/10497929/Dockerfile.txt)

Please let me know if any details are required.

**Comment by surajgawas110** [comment]
Its been two weeks, what's the latest on this issue?

**Comment by ansgarm** [comment]
Hi @surajgawas110 👋

Thank you for pinging us again, this issue must've fallen through the cracks somehow.

Just to ensure this issue is stemming from the `deploy/destroy` commands: Running only `cdktf synth` does not cause new Zombie processes to appear, correct?

Also thank you for supplying the Dockerfile to reproduce this, I'll give it a try.

**Comment by surajgawas110** [comment]
> 
Not sure about synth command behaviour, as I tried synth but it failed as shown below, further 'cdktf get' was also not getting executed. I ran 'cdktf get' only once while initializing the project to fetch all the aws provider files, then after that we just use deploy and destroy itself.

![image](https://user-images.githubusercontent.com/48012652/218381016-2e5db66b-bab6-412e-93db-6ae1d02e69ac.png)

Let me know if anything else is required.

**Comment by ansgarm** [comment]
Could you try upgrading your `cdktf` dependency in your `pom.xml` to match the version of the cdktf-cli (=`0.15.1`) you are using?

**Comment by surajgawas110** [comment]
Hi @ansgarm,

Sorry for the delay. I retested again with changing cdktf dependency version to matching 0.15.1 as you suggested. After this i did "cdktf get" and ran Synth, Deploy and Destroy one after another. For all three cmds I dont see Zombie process being created!
**_So this issue was due to version mismatch between cdktf dependency and cdktf-cli version._**

After **cdktf synth** - No zombie process
![image](https://user-images.githubusercontent.com/48012652/220613863-7fe13691-9ea5-4f33-bbd9-67322720d324.png)

After **cdktf deploy** - No zombie process
![image](https://user-images.githubusercontent.com/48012652/220614078-aa0f4317-fe43-49c7-a3c3-214418cbf530.png)

After **cdktf destroy**- No zombie process
![image](https://user-images.githubusercontent.com/48012652/220614143-eabe03ae-f642-4444-80b2-18f9106ea488.png)

We can close this issue.

**Comment by ansgarm** [comment]
Thanks for the update @surajgawas110 👍
Will close this, if you (or anyone) else encounters this again in the future, don't hesitate to create a new issue!
